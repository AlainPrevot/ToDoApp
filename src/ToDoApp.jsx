import { AppRouter } from "./router/AppRouter";
import "./ToDoApp/event/handleAddEventListener";

export const ToDoApp = () => {
  return (
    <>
      <AppRouter />
    </>
  );
};
