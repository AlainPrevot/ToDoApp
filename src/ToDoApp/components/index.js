export * from './ModeStyle.jsx'
export * from './ToDoAdd.jsx'
export * from './ToDoContentListAdd.jsx'
export * from './ToDoDate.jsx'
export * from './ToDoItem'
export * from './ToDoList.jsx'
export * from './ToDoNav.jsx'
export * from './ToDoOptions.jsx'
export * from './ToDoOptionsButton.jsx'
export * from './ToDoOptionsDropdown.jsx'
