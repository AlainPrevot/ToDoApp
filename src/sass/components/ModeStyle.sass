#mode
    visibility: hidden

.modeDark, .modeLight
    background-position: center
    background-repeat: no-repeat
    background-size: cover
    cursor: pointer
    height: 36px
    position: fixed
    right: 18px
    top: 14px
    width: 36px

.modeLight
    background-image: url('../../Icons/Sol.png')
    display: none

.modeDark
    background-image: url('../../Icons/Luna.png')

#mode:checked ~ .modeLight
    display: block

#mode:checked ~ .modeDark
    display: none
