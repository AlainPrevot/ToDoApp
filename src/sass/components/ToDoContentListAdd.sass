.ToDoContentListAdd
    width: 90%
    height: 70%
    position: relative

.toDoContentListAdd__div
    height: 90%
    overflow-y: scroll

.toDoContentListAdd__div::-webkit-scrollbar
    -webkit-appearance: none

.toDoContentListAdd__div::-webkit-scrollbar:vertical
    width: 4px

.toDoContentListAdd__div::-webkit-scrollbar:horizontal
    height: 2px

.toDoContentListAdd__div::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoContentListAdd__div::-webkit-scrollbar-track
    border-radius: 10px

.toDoContentListAdd__div::-webkit-scrollbar-button:increment, .toDoContentListAdd__div::-webkit-scrollbar-button
    display: none

@media only screen and (min-width: 560px)

    .toDoContentListAdd__div
        height: 80%
