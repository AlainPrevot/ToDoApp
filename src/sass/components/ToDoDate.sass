.toDoDate
    margin: 50px 0
    text-align: center

    h2
        color: white
        font-size: 2.8rem

    h6
        color: #6E9CB9
        font-size: 1.2rem
        font-weight: 300
        margin: -0.7rem
