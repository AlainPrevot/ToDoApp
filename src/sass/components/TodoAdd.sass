
.toDoAdd__input
    background-color: #151D27
    border-radius: 6px
    border: none
    color: white
    height: 46px
    margin-bottom: 20px
    outline: none
    padding-left: 38px
    width: 100%

.toDoAdd__input::placeholder
    color: #6E9CB9
