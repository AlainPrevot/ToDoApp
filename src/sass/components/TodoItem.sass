.toDoItem
    cursor: pointer
    font-size: 1.8rem
    margin-left: 8px

.toDoItem--realizado
    color: var(--Color__primary)

    .toDoItem__span
        text-decoration: line-through

.toDoItem__span
    align-items: center
    display: flex
    font-size: 1rem
    height: 100%
    position: relative
    text-transform: capitalize
    top: -3px
