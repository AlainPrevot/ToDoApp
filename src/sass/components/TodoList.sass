.toDoList
    height: 400px
    overflow-x: hidden
    overflow-y: auto

.toDoList::-webkit-scrollbar
    -webkit-appearance: none

.toDoList::-webkit-scrollbar:vertical
    width: 4px

.toDoList::-webkit-scrollbar:horizontal
    height: 2px

.toDoList::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoList::-webkit-scrollbar-track
    border-radius: 10px

.toDoList::-webkit-scrollbar-button:increment, .toDoList::-webkit-scrollbar-button
    display: none
