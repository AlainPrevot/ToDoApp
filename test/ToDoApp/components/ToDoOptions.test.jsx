import { fireEvent, render } from "@testing-library/react";
import { ToDoOptions } from "../../../src/ToDoApp/components";

/* eslint-disable no-undef */
describe("Pruebas en <ToDoOptions/>", () => {
  test("Debe de hacer match con el Snapshot", () => {
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe de dibujar el button", () => {
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
      />
    );

    const button = container.querySelector("button");

    expect(button).not.toBeNull();
  });

  test("Debe de llamar a la funcion onDeleteTodo y enviar el id del todo", () => {
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
      />
    );

    const button = container.querySelector("button");

    button.click();

    expect(onDeleteTodo).toHaveBeenCalledTimes(1);
  });

  test("Debe de agregar la clase visible al div que contenga la clase menu", () => {
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
      />
    );

    const button = container.querySelector(".ToDoOptionsDropdown__box");

    fireEvent.click(button);

    const menu = container.querySelector(".toDoOptions__box");

    expect(menu.classList.contains("visible")).toBe(true);
  });

  test("debe de eliminar la clase visible del div si ya la contiene", () => {
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
      />
    );

    const button = container.querySelector(".ToDoOptionsDropdown__box");

    fireEvent.click(button);

    const menu = container.querySelector(".toDoOptions__box");

    fireEvent.click(button);

    expect(menu.classList.contains("visible")).toBe(false);
  });

  test("Debe de dibujarse un ToDoOptionsButton por cada elemento del array, menos el que tenga el mismo path", () => {
    Object.defineProperty(window, "location", {
      value: {
        pathname: "/day",
      },
      writable: true,
    });
    const todo = {
      id: 1,
      done: false,
      description: "Aprender React",
    };
    const onDeleteTodo = jest.fn();
    const onTransferTodo = jest.fn();
    const array = ["Day", "Week", "Month", "Year"];
    const { container } = render(
      <ToDoOptions
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onTransferTodo={onTransferTodo}
        array={array}
      />
    );

    const buttons = container.querySelectorAll(".ToDoOptionsButton");

    expect(buttons.length).toBe(3);
  });
});
