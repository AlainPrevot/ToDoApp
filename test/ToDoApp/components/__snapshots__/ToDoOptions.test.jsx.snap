// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ToDoOptions/> Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="toDoOptions"
  >
    <div
      class="ToDoOptionsDropdown__box menu"
    >
      <div
        class="ToDoOptionsDropdown__box--icons menu"
      />
      <div
        class="ToDoOptionsDropdown__box--icons menu"
      />
      <div
        class="ToDoOptionsDropdown__box--icons menu"
      />
    </div>
    <div
      class="toDoOptions__box menu"
      id="1"
    >
      <button
        class="danger"
      >
        Delete
      </button>
      <span
        class="menu"
      >
        Transferir
      </span>
      <ul
        class="toDoOptions__box--addresses"
      >
        <li
          class="ToDoOptionsButton"
        >
          <button>
            Day
          </button>
        </li>
        <li
          class="ToDoOptionsButton"
        >
          <button>
            Week
          </button>
        </li>
        <li
          class="ToDoOptionsButton"
        >
          <button>
            Month
          </button>
        </li>
        <li
          class="ToDoOptionsButton"
        >
          <button>
            Year
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
